### DISCUSSION TITLE
Build should not fail if no snapshots when using PERCY_PARTIAL_BUILD

### DISCUSSION THREAD

**Discussion by yjaaidi** [percy/cli#1013] [Ideas]
## The problem

There are some incremental build/test workflows where no snapshots are taken.
In these cases, `percy exec` or `percy exec:start` & `percy exec:stop` are used in conjunction with `PERCY_PARTIAL_BUILD=1` but as there is no snapshot taken, the build fails.

In a more precise example, we are using [Nx](https://nx.dev/) which has an `nx affected` command that only runs affected tests.
Some commits or PRs might change files that don't affect tests.
In this case, running the following command would fail as no snapshot is uploaded:
```sh
PERCY_PATIAL_BUILD=1 percy exec nx affected --target test
```

## Solutions

1. Allow empty builds when using `PERCY_PARTIAL_BUILD`
2. Add a `PERCY_ALLOW_EMPTY_BUILD` env var that sends an `allow_empty_build: true` flag to the API when creating the build and mark the build as successful on finalize even if no snapshot was upload.

## Workarounds

1. take a placeholder snapshot no matter what (cf. https://github.com/marmicode/marmicode/blob/59412b8b1d721ba9fc461b38605f4b8bec2f3cc0/.github/workflows/test-and-deploy.yml#L96 & https://github.com/marmicode/marmicode/blob/main/tools/percy-upload-placeholder.ts)
3. guess if some snapshots will be taken or not and skip starting percy (but this will not add a percy check to the PR, which might make it unmergeable if the target branch is protected)
4. run all tests which defeats the whole purpose of incremental build/test tools like Nx

**Comment by Robdel12** [comment]
Hey @yjaaidi! Thanks for the idea, I’ve let the product team know about it.

**Reply by adbutterfield** [reply]
I just hit the same exact problem...
